src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -155,6 +159,59 @@ public static string GetGlobalPackagesFolder(ISettings settings)
             return path;
         }
 
+        /// <summary>
+        /// Read fallback folders from the environment variable or from nuget.config.
+        /// </summary>
+        public static IReadOnlyList<string> GetFallbackPackageFolders(ISettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            var paths = new List<string>();
+
+            var envValue = Environment.GetEnvironmentVariable(FallbackPackagesFolderEnvironmentKey);
+
+            if (string.IsNullOrEmpty(envValue))
+            {
+                // read config values
+                paths.AddRange(GetFallbackPackageFoldersFromConfig(settings));
+            }
+            else
+            {
+                paths.AddRange(envValue.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries));
```

Comment:
I have a bit of code that determines whether to show stack or not, this has an mockable environment variable reader thingy. That would allow testing.

---
In reply to: [66689689](https://github.com/NuGet/NuGet.Client/pull/673#discussion_r66689689) [](ancestors = 66689689)